tools/export_model_images.py

```diff
@@ -76,62 +77,57 @@ def export_images(provider: str, kinds: list):
+            lines = takewhile(lambda l: not l.startswith("## "), file.readlines())
     else:
+        # provider file does not exist, create default header
         lines = [
-            f"---\nsidebar_label: {provider.capitalize()}\n---",
-            "",
-            f"# {provider.capitalize()} Resource Data Models",
-            "",
-            "```mdx-code-block\nimport ZoomPanPinch from '@site/src/components/ZoomPanPinch';\n```",
+            f"---\nsidebar_label: {provider.capitalize()}\n---\n\n",
+            f"# {provider.capitalize()} Resource Data Models\n\n",
+            "```mdx-code-block\nimport ZoomPanPinch from '@site/src/components/ZoomPanPinch';\n```\n\n",
         ]
 
     with open(f"./{provider}/index.md", "w+") as file:
-        original_stdout = sys.stdout
-        sys.stdout = file
-
-        if ":::\n" in lines:
-            lastline = ":::"
-        else:
-            lastline = "```"
-
         for line in lines:
-            line = line.strip("\n")
-            print(line)
-
-            if line == lastline:
-                print()
-                break
+            file.write(line)
 
         for name in sorted(a["fqn"] for a in kinds):
-            print(f"## `{name}`\n")
-            print(f"<ZoomPanPinch>\n\n![Diagram of {name} data model](./img/{name}.svg)\n\n</ZoomPanPinch>\n")
-            print(f"<details>\n<summary>Relationships to Other Resources</summary>\n<div>\n<ZoomPanPinch>\n")
-            print(f"![Diagram of {name} resource relationships](./img/{name}_relationships.svg)\n")
-            print(f"</ZoomPanPinch>\n</div>\n</details>\n")
-
-        sys.stdout = original_stdout
-
-
-kinds = get_kinds()
-retries = 0
-print(f"Kinds: {list(kinds.keys())}")
-
-while retries < 60 and not required_providers.issubset(list(kinds.keys())):
-    print(f"Missing some required kinds, trying again in 5 seconds...")
-    time.sleep(5)
-    kinds = get_kinds()
-    print(f"New kinds: {list(kinds.keys())}")
-    retries += 1
-
-for provider in providers:
-    if len(kinds[provider]) > 0:
-        if not os.path.exists(f"./{provider}"):
-            os.mkdir(f"./{provider}")
-        if not os.path.exists(f"./{provider}/img"):
-            os.mkdir(f"./{provider}/img")
-        export_images(provider, kinds[provider])
-        print_md(provider, kinds[provider])
+            file.write(f"## `{name}`\n\n")
+            file.write(f"<ZoomPanPinch>\n\n![Diagram of {name} data model](./img/{name}.svg)\n\n</ZoomPanPinch>\n\n")
+            file.write(f"<details>\n<summary>Relationships to Other Resources</summary>\n<div>\n<ZoomPanPinch>\n\n")
+            file.write(f"![Diagram of {name} resource relationships](./img/{name}_relationships.svg)\n\n")
+            file.write(f"</ZoomPanPinch>\n</div>\n</details>\n\n")
+
+
+def load_valid_kinds() -> Dict[str, Any]:
+    for _ in range(30): # nr of retries
+        try:
+            print("Get available kinds...")
+            kinds = get_kinds()
+            if kinds:
```

Comment:
We need to check if the returned kinds are a subset of `required_providers`, as there are times when Resoto Core takes a long time to start up in the CI. This check was added previously since I witnessed it beginning the import before all kinds had been loaded.